torch/_ops.py

```diff
@@ -26,9 +32,127 @@ def dl_open_guard():
         sys.setdlopenflags(old_flags)
 
 
+# TODO(voz) We are missing an entire axis of registration - Modes for the python key
+class PyOperatorABC(ABC):
+    def __call__(self, *args, **kwargs):
+        pass
+
+    def py_impl(self, dispatch_key, fn):
+        pass
+
+    def name(self):
+        pass
+
+    def dispatch(self, dispatch_key, *args, **kwargs):
+        pass
+
+
+class PyOperator(PyOperatorABC):
+    def __init__(self, name):
+        self._name = name
+        self.table = {}
+        self.python_key_mode_table = {}
+
+        # Make _OPNamespace not scream, this whole name based association needs a good hard look
+        self.__name__ = "pyop." + name
```

Comment:
TBH, there isn't really any good reason to make pyops available at torch.ops.pyop. It is necessary for c++ bound ops as we must introduce them to Python namespace somewhere, but for Python ops you can use the normal namespacing mechanisms of Python to handle their namespacing

Reply:
This is to hack away at the first getattr. Removing this "pyop." addition gives us:

```
TypeError: '_OpNamespace' object is not callable
```
Hence the vague comment. Its a bit of a huge hack to get around this:

```
    def __getattr__(self, name):
        # Here we are creating `torch.ops.my_namespace`
        namespace = _OpNamespace(name)
        setattr(self, name, namespace)
        return namespace
```

Alternatively, we could keep a global list to check, and if the op is there, skip the namespace notion, or have a "default" namespace of sorts we consult first...